plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/layout/SubsystemLayout.java

```diff
@@ -51,21 +54,50 @@
   private Subscription retained; //NOPMD field due to GC
   private DataSource<?> source;
 
+  private List<Component> hidden = new ArrayList<>();
+
   @FXML
   private void initialize() {
     retained = EasyBind.listBind(container.getChildren(), EasyBind.map(children, this::paneFor));
+
+    ActionList.registerSupplier(container, () -> {
+      ActionList al = ActionList.withName(getTitle());
+      if (!hidden.isEmpty()) {
+        al.addAction("Unhide all", () -> {
+          children.addAll(hidden);
+          hidden.clear();
+        });
+      }
+      return al;
+    });
   }
 
+  private WeakHashMap<Component, Pane> panes = new WeakHashMap<>();
+
   private Pane paneFor(Component component) {
+    if (panes.containsKey(component)) {
+      return panes.get(component);
```

Comment:
this pattern (EasyBind.map -> WeakHashMap -> func) appears a couple of places in the code, maybe should extract a util?

Reply:
SubsystemLayout and ListLayout should really inherit from a common superclass that keeps all this code in the same place.